### PR TITLE
[WFCORE-5324] Support for CLI script executing at boot for bootable JAR

### PR DESCRIPTION
**Pull request by jfdenise**
Issues:
https://issues.redhat.com/browse/WFCORE-5322
https://issues.redhat.com/browse/WFCORE-5323
https://issues.redhat.com/browse/WFCORE-5324

Community doc PR:
https://github.com/wildfly/wildfly/pull/14069

* Introduce --cli-script=[path to cli script]
* Fix output when error occur during CLI script execution
* Fix ability to set server log config directory
* Evolved ts.bootable profile of elytron tests with CLI configuration applied at runtime.

### CODE REVIEW

File: bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Arguments.java [hunk 1]
```diff
@@ -101,6 +102,11 @@ private void handleArguments(List<String> args) throws Exception {
                 serverArguments.add(a);
             } else if (CommandLineConstants.HELP.equals(a)) {
                 isHelp = true;
+            } else if (a.startsWith("--cli-script")) {
+                cliScript = Paths.get(getValue(a));
```
Comment (jmesnil) [inline bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Arguments.java:106]: what would happen if the path of the script is relative? is it resolved based on the bootable jar path or some of our well known dirs?

File: bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java [hunk 2]
```diff
@@ -230,6 +230,17 @@ private LogContext configureLogContext() throws IOException {
     }
 
     public void run() throws Exception {
+        Path script = arguments.getCLIScript();
+        if (script != null) {
+            Path markerDir = environment.getJBossHome().resolve("cli-boot-hook-dir");
+            Path outputFile = environment.getJBossHome().resolve("cli-boot-hook-output-file.txt");
+            Files.createDirectories(markerDir);
```
Comment (jmesnil) [inline bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java:237]: being consistent with S2I is more important but we should avoid creating these types of dirs directly in `$JBOSS_HOME`. the `data` directory is a better location for such resources.

File: bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/_private/BootableJarLogger.java [hunk 3]
```diff
@@ -162,4 +162,7 @@
 
     @Message(id = Message.NONE, value = "Display the content of the Galleon configuration used to build this bootable JAR")
     String argDisplayGalleonConfig();
+
+    @Message(id = Message.NONE, value = "Path to a CLI script to execute")
```
Comment (jmesnil) [inline bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/_private/BootableJarLogger.java:166]: maybe specify that this option is when running the bootable jar (to avoid being confused with CLI scripts that are executed when the jar is built).

File: cli/src/main/java/org/jboss/as/cli/impl/BootScriptInvoker.java [hunk 4]
```diff
@@ -87,11 +88,6 @@ public void runCliScript(ModelControllerClient client, File file) {
         } catch (Exception ex) {
             try {
                 ROOT_LOGGER.errorProcessingScript(file);
-                StringBuilder errBuilder = new StringBuilder();
```
Comment (jmesnil) [inline cli/src/main/java/org/jboss/as/cli/impl/BootScriptInvoker.java:90]: Is it intentional to get rid of this?

File: cli/src/main/java/org/jboss/as/cli/impl/CliConfigImpl.java [hunk 5]
```diff
@@ -94,6 +94,7 @@
     static CliConfig newBootConfig() throws CliInitializationException {
         CliConfigImpl config = new CliConfigImpl();
         config.validateOperationRequests = false;
+        config.echoCommand = true;
```
Comment (jmesnil) [inline cli/src/main/java/org/jboss/as/cli/impl/CliConfigImpl.java:97]: is this change required to support the new `--cli-script`?

File: bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Arguments.java (hunk 1 above)
Comment (jfdenise) [reply bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Arguments.java:106]: If relative, it is relative to the JVM work dir (from where the JVM has been launched).

File: bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java (hunk 2 above)
Comment (jfdenise) [reply bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java:237]: @jmesnil , you mean <home>/standalone/data? Wondering if <home>/standalone/tmp would be more appropriate?

File: bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/_private/BootableJarLogger.java (hunk 3 above)
Comment (jfdenise) [reply bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/_private/BootableJarLogger.java:166]: Good point!

File: cli/src/main/java/org/jboss/as/cli/impl/BootScriptInvoker.java (hunk 4 above)
Comment (jfdenise) [reply cli/src/main/java/org/jboss/as/cli/impl/BootScriptInvoker.java:90]: Yes, the way errors were reported was not nice. The command is now printed to the output and output displayed when an error occurs. Much easier to match the faulty line and the error. The output is not shown if the CLI script execution was successful.

File: cli/src/main/java/org/jboss/as/cli/impl/CliConfigImpl.java (hunk 5 above)
Comment (jfdenise) [reply cli/src/main/java/org/jboss/as/cli/impl/CliConfigImpl.java:97]: No, not required. The error reporting is now better.

**Review by jmesnil — APPROVED** [review]

**Comment by jmesnil** [comment]
@jfdenise thanks!
